src/MomentoTaggedCache.php

```diff
@@ -27,21 +34,51 @@ public function put($key, $value, $ttl = null)
     public function get($key, $default = null): mixed
     {
         $tags = $this->tags->getNames();
+        if (!self::validateTags($tags)) {
+            return false;
+        }
         $cacheName = $this->store->getCacheName();
         $value = null;
+        $newKey = self::createNewKey($tags, $key);
+        $hashedKey = hash("sha256", $newKey);
         foreach ($tags as $tag) {
```

Comment:
My understanding is this:
In case of `Cache::tags(['people', 'artists'])->put('John', $john, $seconds);`, we need to create two sets. 
One is `people` as a set name and the other is `artists` as a set name.
Each will have a hashed version of `people-artists-John`.
And we store the hashed key with the value `$john`.

I think we need to keep track of all tags since users are able to flush all values associated with each tag with `Cache::tags(['people', 'authors'])->flush();`.
In this case, we need all keys from both tags `people` and `authors` to `forget` values associated with those keys in both sets.

Reply:
Chris is right. This is a `get`, and we already have all the information we need to retrieve the value from the cache: the hashed key. This can be simplified down to using `$this->store->get()` to try to get the value from the cache using the hashed key. If we get a value back, we return it. If we don't, we return `$default`. We only ever need to use the sets when handling `put` and `flush`.

Reply:
Ah! Somehow I thought it was for `put`. Yes, you're both right about not needing the for loop.